src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,137 @@
 package core.basesyntax;
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final double LOAD_FACTOR = 0.75;
+    private int arraySize = 16;
+    private int size = 0;
+    private Node<K, V>[] array;
+
+    public MyHashMap() {
+        array = new Node[arraySize];
+    }
 
     @Override
     public void put(K key, V value) {
-
+        if (size == (int) (arraySize * LOAD_FACTOR)) {
+            resizeArray();
+        }
+        if (key == null) {
+            putNullKey(value);
+            return;
+        }
+        int index = getIndex(key);
+        Node<K, V> newNode = new Node<>(key, value, null);
+        if (array[index] == null) {
+            array[index] = newNode;
+        } else {
+            Node<K, V> previousNode = null;
+            Node<K, V> currentNode = array[index];
+            while (currentNode != null) {
+                if (currentNode.key.equals(key)) {
+                    currentNode.value = value;
+                    return;
+                }
+                previousNode = currentNode;
+                currentNode = currentNode.next;
+            }
+            if (previousNode != null) {
+                previousNode.next = newNode;
+            }
+        }
```

Comment:
let's rewrite this block of code on something like this, if you'll have questions tag me): 
Node nodeByIndex = table[index];
while (node != null) {
  if (keys equal) {
     return;
  }
 if (node.next == null) {
   node.next = newNode;
   return;
 }
}
table[index] = newNode;